cachi2/core/package_managers/yarn/utils.py

```diff
@@ -27,3 +29,68 @@ def run_yarn_cmd(
     except subprocess.CalledProcessError as e:
         # the yarn command writes the errors to stdout
         raise PackageManagerError(f"Yarn command failed: {' '.join(cmd)}", stderr=e.stdout)
+
+
+VersionLike = Union[Version, str]
+
+
+class VersionsRange:
+    """Represents a version range for cleaner version constrains checks.
+
+    Versions range is a right-open interval:
+    >>> Version.parse("1.2.3") in VersionsRange("3.0.0", "4.0.0")
+    False
+    >>> Version.parse("1.2.3") in VersionsRange("1.0.0", "2.0.0")
+    True
+    >>> Version.parse("1.0.0") in VersionsRange("1.0.0", "2.0.0")
+    True
+    >>> Version.parse("2.0.0") in VersionsRange("1.0.0", "2.0.0")
+    False
+
+    Release candidates are a special case, they are ignored within the
+    interval and cause immediate rejection on any of the boundaries:
+    >>> Version.parse("2.0.0-rc1") in VersionsRange("1.0.0", "2.0.0")
+    False
+    >>> Version.parse("1.0.0-rc1") in VersionsRange("1.0.0", "2.0.0")
+    False
+    >>> Version.parse("1.5.0-rc1") in VersionsRange("1.0.0", "2.0.0")
+    True
+    """
+
+    def __init__(self, min_ver: VersionLike, max_ver: VersionLike) -> None:
+        """Initialize a version range."""
+        self.min_ver = min_ver if isinstance(min_ver, Version) else Version.parse(min_ver)
+        self.max_ver = max_ver if isinstance(max_ver, Version) else Version.parse(max_ver)
+
+    def __contains__(self, other: Version) -> bool:
+        if not isinstance(other, self.min_ver.__class__):
+            return False
+        # The original version check logic (as captured in UTs) broke with direct
+        # version comparison rules, e.g. 4.0.0.rc1 would have been considered
+        # version 4 and would  have been rejected basing on major version
```

Comment:
This was intentional and I'd argue that it's even correct since RCs are like ~90% of the final release and so if we're not ready for the next major we can't be ready even for the the RC in terms of support, so how exactly did it break UTs?